### PR TITLE
Add power-assert enabled t.assert() assertion; remove from others

### PR DESCRIPTION
**Pull request by eemed**
Initial for #1808 . I still have those 3 failing tests but could you @novemberborn take a look at this first.
I moved these functions away from enhanced assert.
-    t.truthy()
-    t.falsy()
-    t.true()
-    t.false()
-    t.regex()
-    t.notRegex()

Added a new function t.assert() to enhanced assert and copied over the t.truthy() tests.
Then I tried to update the tests that kept failing. I also updated the patterns in babel-preset-transform-test-files.

### CODE REVIEW

File: test/assert.js [hunk 1]
```diff
@@ -1505,6 +1505,7 @@ test('.regex() fails if passed a bad value', t => {
 		assertions.regex(42, /foo/);
 	}, {
 		assertion: 'regex',
+		imporoperUsage: true,
```
Comment (novemberborn) [inline test/assert.js]: There's a typo here.

File: test/fixture/enhanced-assertion-formatting.js [hunk 2]
```diff
@@ -4,16 +4,16 @@ const foo = 'foo';
...
 });
 
-test('fails with "instanceof" expression', t => {
-	// eslint-disable-next-line no-new-object
-	t.false(new Object(foo) instanceof Object);
-});
+// test('fails with "instanceof" expression', t => {
+// 	// eslint-disable-next-line no-new-object
+// 	t.false(new Object(foo) instanceof Object);
+// });
```
Comment (novemberborn) [inline test/fixture/enhanced-assertion-formatting.js]: Let's keep this test case, but then with `t.assert()`.

File: test/fixture/report/regular/test.js [hunk 3]
```diff
@@ -26,8 +26,8 @@ test('formatted', t => {
 });
 
 test('power-assert', t => {
-	const foo = 'bar';
-	t.falsy(foo);
+	const foo = '';
+	t.assert(foo);
```
Comment (chocolateboy) [inline test/fixture/report/regular/test.js]: To match the original test, shouldn't this be:

```js
t.assert(!foo);
```

**Comment by eemed** [comment]
Updated tests and used .assert() in the test case. PR for babel-preset-transform-files is [here.](https://github.com/avajs/babel-preset-transform-test-files/pull/15)

File: test/fixture/report/regular/test.js (hunk 3 above)
Comment (eemed) [reply test/fixture/report/regular/test.js]: Don't really know. Is it important that they match?

Comment (novemberborn) [reply test/fixture/report/regular/test.js]: I guess @chocolateboy's point is that you also changed the value of `foo`. But, in the end, the test still fails, so it doesn't change too much.

**Comment by novemberborn** [comment]
Looks great @eemed. Would you want to have a stab at updating the documentation?

**Comment by eemed** [comment]
I can take a look. Do I just commit doc updates to this pr aswell?

**Comment by novemberborn** [comment]
> Do I just commit doc updates to this pr aswell?

Yep!

**Comment by eemed** [comment]
I'm not sure if there's more updating to be done. I tried to look through all the files but didn't find anything else.

**Comment by novemberborn** [comment]
@eemed I've pushed some documentation updates, let me know what you think.

Let's hold off merging until we can address https://github.com/avajs/eslint-plugin-ava/issues/218. Perhaps you'd like to tackle that one too? 😄

**Comment by eemed** [comment]
I can take a look at it this weekend.

**Pull request merged**